test/lhttpc_manager_tests.erl

```diff
@@ -300,13 +298,27 @@ closed_race_cond() ->
     ManagerPid = whereis(lhttpc_manager),
     true = erlang:suspend_process(ManagerPid),
 
+    case CloseBeforeDispatch of
+        true ->
+            gen_tcp:close(LS), % a closed message should be sent to the manager
+            timer:sleep(1000);
```

Comment:
I assumed it's for testing a race condition or letting the system reach some eventually consistent state... It's a unit test, so not particularly egregious.